aten/src/ATen/native/NativeFunctions.cpp

```diff
@@ -584,5 +586,77 @@ Tensor RoiPooling2d_backward_cpu(
   throw std::runtime_error("not implemented");
 }
 
+
+
+// TODO Replace this with more accurate digamma().
+template <typename Scalar>
+static inline Scalar digamma_one(Scalar x) {
+  const Scalar eps = x * 1e-2;
+  return (std::lgamma(x + eps) - std::lgamma(x - eps)) / (eps + eps);
+}
+
+/** Computes the reparameterized gradient -(d/dalpha cdf(x;alpha)) / pdf(x;alpha)
+    for random number x drawn from a standard Gamma distribution Gamma(alpha).
+*/
+template <typename Scalar>
```

Comment:
Scalar is the name of a concrete type in ATen. Let's not reuse the same name.

Reply:
ok, I'm going to go with CScalar, matching toCFloat, toCDouble, etc.